cmake/boost.cmake

```diff
@@ -0,0 +1,66 @@
+# Code to import Boost correctly, either from the system, or from a package.
+
+find_package(Boost 1.74 COMPONENTS log)
+
+option(MENDER_DOWNLOAD_BOOST "Download Boost if it is not found (Default: OFF)" OFF)
+
+if(MENDER_DOWNLOAD_BOOST AND NOT ${Boost_FOUND})
+  FetchContent_Declare(
+    Boost
+    # SYSTEM is only supported in CMake 3.25 and later, but is necessary in order to exclude Boost
+    # from our restrictive warnings-as-errors, in particular -Wsuggest-override. See workaround
+    # below.
+    SYSTEM
+    DOWNLOAD_EXTRACT_TIMESTAMP ON
+    URL https://github.com/boostorg/boost/releases/download/boost-1.83.0/boost-1.83.0.tar.gz
+    URL_MD5 58db882403e0c16b334760f3c3b76ff8
```

Comment:
This hurts. The network layers used here have integrity checks so we don't need to duplicate that. And this has nothing to do with a security check. So please either drop it or replace it with a better checksum algorithm.

Reply:
Fixed.